src/main/java/mate/academy/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,47 @@
+package mate.academy.dao.impl;
+
+import jakarta.persistence.Query;
+import java.util.List;
+import mate.academy.dao.OrderDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.Order;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+public class OrderDaoImpl implements OrderDao {
+    @Override
+    public List<Order> getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query query = session.createQuery("SELECT o FROM Order o WHERE o.user = :user",
+                    Order.class);
+            query.setParameter("user", user);
+            return query.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can not find order by user " + user, e);
+        }
+    }
+
+    @Override
+    public Order add(Order order) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(order);
```

Comment:
Consider using `session.save(order)` instead of `session.persist(order)`. While both methods are used to save an entity, `save()` returns the generated identifier, which might be useful if you need it immediately after saving. Additionally, `save()` is more commonly used in Hibernate-specific implementations.